common/src/main/java/org/conscrypt/HpkeContextRecipient.java

```diff
@@ -25,83 +25,83 @@
+     *                                  does not match the size expectation
+     * @throws IllegalStateException    if an issue is encountered while setting up the recipient
+     *                                  (an issue could occur most likely if the keys configured are
+     *                                  not valid)
+     */
+    public static HpkeContextRecipient setupBase(
+            HpkeSuite hpkeSuite, byte[] enc, PrivateKey privateKey, byte[] info) {
+        Preconditions.checkNotNull(hpkeSuite, "hpkeSuite");
+        hpkeSuite.getKem().validateEncLength(enc);
+        final byte[] sk = hpkeSuite.getKem().validatePrivateKeyTypeAndGetRawKey(privateKey);
+        try {
+            final HpkeContextRecipient ctxRecipient = new HpkeContextRecipient(hpkeSuite);
+            ctxRecipient.ctx = (EVP_HPKE_CTX) NativeCrypto.EVP_HPKE_CTX_setup_recipient(
+                    hpkeSuite.getKem().getId(), hpkeSuite.getKdf().getId(),
+                    hpkeSuite.getAead().getId(), sk, enc, info);
+            return ctxRecipient;
+        } catch (Exception e) {
+            throw new IllegalStateException(
+                    "Error while setting up base recipient with the keys provided", e);
+        }
     }
-  }
 
-  /**
-   * Hybrid Public Key Encryption (HPKE) decryption.
-   * <p>
-   * Note: This API is stateful.
-   * <p><p>
-   * Multiple message decryption:<p>
-   * If decrypting multiple messages that were encrypted in a sequence using the same context,
-   * they must be decrypted in the same order as well. The method setup[Mode]Recipient must be
-   * called just once before multiple calls to this API are made.
-   * <p><p>
-   * Single message decryption:<p>
-   * If decrypting a single message, the method setup[Mode]Recipient must be called before each
-   * message. Calling setup[Mode]Recipient creates a new HPKE context.
-   *
-   * @param ciphertext contains the encrypted plaintext
-   * @param aad        optional associated data
-   * @return plaintext
-   * @throws IllegalStateException if an issue is encountered while decrypting (an issue could occur
-   *                               most likely if the keys configured are not valid)
-   */
-  public byte[] open(byte[] ciphertext, byte[] aad) {
-    Preconditions.checkNotNull(ciphertext, "ciphertext");
-    try {
-      return NativeCrypto.EVP_HPKE_CTX_open(ctx, ciphertext, aad);
-    } catch (Exception e) {
-      throw new IllegalStateException(
-          "Error while decrypting with the keys provided during setup recipient", e);
+    /**
+     * Hybrid Public Key Encryption (HPKE) decryption.
+     * <p>
+     * Note: This API is stateful.
+     * <p><p>
+     * Multiple message decryption:<p>
+     * If decrypting multiple messages that were encrypted in a sequence using the same context,
+     * they must be decrypted in the same order as well. The method setup[Mode]Recipient must be
+     * called just once before multiple calls to this API are made.
```

Comment:
I think this last sentence can be dropped now. It's now captured in the type system! :-)

Reply:
Done, I have rephrased the documentation for both